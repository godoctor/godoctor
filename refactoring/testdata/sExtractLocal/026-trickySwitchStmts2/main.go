package main

import (
	"fmt"
	"strconv"
)

func main() {
	x := 2
	y := 5
	var choice string
	fmt.Println("please choose: x + x, x * y?")
	fmt.Scanf(choice)
	iChoice, _ := strconv.Atoi(choice)
	switch iChoice {
	case x + x:
		fmt.Println(x + x)
	case x * y: // <<<<< extractLocal,18,6,18,10,newVar,pass
		fmt.Println(x * y)
	default:
		fmt.Println("didn't work")
	}
}
