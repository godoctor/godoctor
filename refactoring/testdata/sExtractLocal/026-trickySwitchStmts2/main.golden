package main

import (
	"fmt"
	"strconv"
)

func main() {
	x := 2
	y := 5
	var choice string
	fmt.Println("please choose: x + x, x * y?")
	fmt.Scanf(choice)
	iChoice, _ := strconv.Atoi(choice)
	newVar := x * y
	switch iChoice {
	case x + x:
		fmt.Println(x + x)
	case newVar: // <<<<< extractLocal,18,6,18,10,newVar,pass
		fmt.Println(x * y)
	default:
		fmt.Println("didn't work")
	}
}
